frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -4,6 +4,33 @@ import _ from "lodash";
 import type { Action, State } from "./hello-world";
 import { deriveSwitchStatus, Group } from "./hello-world";
 
+// handle custom ; 4th param will be project state object
+const updateGroupstate = (
+  state: State,
+  group: Group,
+  project: string,
+  isCustom?: boolean
```

Comment:
@danielhochman I know you suggested using the ProjectState object but the interface requires the "checked" arg & since what we want to account for is `custom`, I set a bool that the can be passed in. lmk what you think

```
         // user owned project vs custom project
          if (v.from.users.length > 0) {
            newState = updateGroupstate(newState, Group.PROJECTS, k, false);
          } else if (v.from.selected) {
            newState = updateGroupstate(newState, Group.PROJECTS, k, true);
         }
```

Reply:
i think we also need to be able to add a project with checked = false though 

e.g. https://github.com/lyft/clutch/pull/1465/files#diff-66c15f7738bc8884141f5f91e2c8444c958af9bd00c50832621795772b0f8e53R122-R128

Reply:
this is it for upstreams/downstreams (if they're not already in the group) https://github.com/lyft/clutch/pull/1566/files#diff-66c15f7738bc8884141f5f91e2c8444c958af9bd00c50832621795772b0f8e53R24-R28


Group.Projects are checled true if they're not already in the group https://github.com/lyft/clutch/pull/1566/files#diff-66c15f7738bc8884141f5f91e2c8444c958af9bd00c50832621795772b0f8e53R21-R23

Reply:
i think it would be clearer to keep that logic in the reducer